src/translate.py

```diff
@@ -3841,23 +4140,100 @@ def process_instr(instr: Instruction) -> None:
+            reg, expr_fn = arch.instrs_implicit_destination[mnemonic]
+            set_reg(reg, expr_fn(args))
+
+        elif mnemonic in arch.instrs_ppc_compare:
+            set_reg(Register("cr0_eq"), arch.instrs_ppc_compare[mnemonic](args, "=="))
+            set_reg(Register("cr0_gt"), arch.instrs_ppc_compare[mnemonic](args, ">"))
+            set_reg(Register("cr0_lt"), arch.instrs_ppc_compare[mnemonic](args, "<"))
+            set_reg(Register("cr0_so"), Literal(0))
+
         elif mnemonic in arch.instrs_no_dest:
             stmt = arch.instrs_no_dest[mnemonic](args)
             to_write.append(stmt)
 
-        elif mnemonic in arch.instrs_destination_first:
+        elif mnemonic.rstrip(".") in arch.instrs_destination_first:
             target = args.reg_ref(0)
-            val = arch.instrs_destination_first[mnemonic](args)
+            val = arch.instrs_destination_first[mnemonic.rstrip(".")](args)
             # TODO: IDO tends to keep variables within single registers. Thus,
             # if source = target, maybe we could overwrite that variable instead
             # of creating a new one?
             set_reg(target, val)
             mn_parts = mnemonic.split(".")
-            if (len(mn_parts) >= 2 and mn_parts[1] == "d") or mnemonic == "ldc1":
+            if mnemonic.endswith("."):
+                # PPC instructions suffixed with . set condition bits (CR0) based on the result value
+                target_reg = args.reg(0)
+                set_reg(
+                    Register("cr0_eq"),
+                    BinaryOp.icmp(target_reg, "==", Literal(0, type=target_reg.type)),
+                )
+                # Use manual casts for cr0_gt/cr0_lt so that the type of target_reg is not modified
+                # until the resulting bit is .use()'d.
+                set_reg(
+                    Register("cr0_gt"),
+                    BinaryOp(
+                        left=Cast(
+                            expr=target_reg,
+                            reinterpret=True,
+                            silent=True,
+                            type=Type.s32(),
+                        ),
+                        op=">",
+                        right=Literal(0),
+                        type=Type.s32(),
+                    ),
+                )
+                set_reg(
+                    Register("cr0_lt"),
+                    BinaryOp(
+                        left=Cast(
+                            expr=target_reg,
+                            reinterpret=True,
+                            silent=True,
+                            type=Type.s32(),
+                        ),
+                        op="<",
+                        right=Literal(0),
+                        type=Type.s32(),
+                    ),
+                )
+                # TODO: Implement overflow checking
+                set_reg(Register("cr0_so"), ErrorExpr(f"overflow of {target}"))
```

Comment:
I'd use a fn_op and remove the TODO. I imagine this never comes up in practice anyway (?).